src/core/output-layout.cpp

```diff
@@ -1577,6 +1578,22 @@ class output_layout_t::impl
         }
     }
 
+    wf::output_t *get_previous_output(wf::output_t *output)
+    {
+        auto os = get_outputs();
+        std::reverse(os.begin(), os.end());
```

Comment:
To me, this reverse step looks unnecessary, we can just take `std::prev(it)` in the normal list.